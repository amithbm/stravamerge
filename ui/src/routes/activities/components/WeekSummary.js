import React from 'react'
import { Row, Col } from 'react-bootstrap'
import { ResponsiveContainer, BarChart, Bar, XAxis, Tooltip } from 'recharts'
import ChartTooltip from '../../common/components/ChartTooltip.js'

export default ({weekData}) => {
    return <Row bsClass="row row-no-padding">
        <Col xs={3} style={{height:"100px"}}>
            <div style={{ display: "table", height: "100px", overflow: "hidden", width: "100%", textAlign: "center" }}>
                <div  style={{display: "table-cell", verticalAlign: "middle"}}>
                    {Math.round(weekData.map(dp => dp.distance).reduce( (p,c) => p+c, 0)*10)/10} km
                </div>
            </div> 
        </Col>
        <Col xs={9} style={{height:"100px"}}>
            <ResponsiveContainer minHeight={100}>
                <BarChart data={weekData}>
                    <XAxis dataKey="label" axisLine={false} tickLine={false} tickFormatter={(s) => s.substring(0,1)} />
                    <Tooltip content={ <ChartTooltip /> } />
                    <Bar dataKey="distance" fill="rgb(4,117,111)" />
                </BarChart>
            </ResponsiveContainer>
        </Col>
    </Row>
}
