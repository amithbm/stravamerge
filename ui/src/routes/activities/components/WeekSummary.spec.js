import React from 'react'
import WeekSummary from './WeekSummary'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<WeekSummary />', () => {
        it('renders as expected', () => {
            const weekData = [
                { distance: 1, label: 'Monday' },
                { distance: 2, label: 'Tuesday' },
                { distance: 3, label: 'Wednesday' },
                { distance: 4, label: 'Thursday' },
                { distance: 5, label: 'Friday' },
                { distance: 6, label: 'Saturday' },
                { distance: 7, label: 'Sunday' },
            ]

            const component = shallow(<WeekSummary weekData={weekData} />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})