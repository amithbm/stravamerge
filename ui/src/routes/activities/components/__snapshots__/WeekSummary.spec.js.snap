exports[`components <WeekSummary /> renders as expected 1`] = `
<Row
  bsClass="row row-no-padding"
  componentClass="div">
  <Col
    bsClass="col"
    componentClass="div"
    style={
      Object {
        "height": "100px",
      }
    }
    xs={3}>
    <div
      style={
        Object {
          "display": "table",
          "height": "100px",
          "overflow": "hidden",
          "textAlign": "center",
          "width": "100%",
        }
      }>
      <div
        style={
          Object {
            "display": "table-cell",
            "verticalAlign": "middle",
          }
        }>
        28
         km
      </div>
    </div>
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    style={
      Object {
        "height": "100px",
      }
    }
    xs={9}>
    <ResponsiveContainer
      debounce={0}
      height="100%"
      minHeight={100}
      width="100%">
      <AnimationDecorator(ComposedDataDecorator(BarChart))
        data={
          Array [
            Object {
              "distance": 1,
              "label": "Monday",
            },
            Object {
              "distance": 2,
              "label": "Tuesday",
            },
            Object {
              "distance": 3,
              "label": "Wednesday",
            },
            Object {
              "distance": 4,
              "label": "Thursday",
            },
            Object {
              "distance": 5,
              "label": "Friday",
            },
            Object {
              "distance": 6,
              "label": "Saturday",
            },
            Object {
              "distance": 7,
              "label": "Sunday",
            },
          ]
        }
        layout="horizontal"
        margin={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        stackOffset="none">
        <XAxis
          allowDataOverflow={false}
          allowDecimals={true}
          axisLine={false}
          dataKey="label"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={30}
          hide={false}
          orientation="bottom"
          padding={
            Object {
              "left": 0,
              "right": 0,
            }
          }
          scale="auto"
          tickCount={5}
          tickFormatter={[Function]}
          tickLine={false}
          type="category"
          width={0}
          xAxisId={0} />
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          content={<ChartTooltip />}
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          isAnimationActive={true}
          itemSorter={[Function]}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          viewBox={
            Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            }
          }
          wrapperStyle={Object {}} />
        <Bar
          animationBegin={0}
          animationDuration={1500}
          animationEasing="ease"
          data={Array []}
          dataKey="distance"
          fill="rgb(4,117,111)"
          isAnimationActive={true}
          layout="vertical"
          legendType="rect"
          minPointSize={0}
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          xAxisId={0}
          yAxisId={0} />
      </AnimationDecorator(ComposedDataDecorator(BarChart))>
    </ResponsiveContainer>
  </Col>
</Row>
`;
